saleor/graphql/product/mutations/products.py

```diff
@@ -1198,106 +1202,122 @@ def perform_mutation(cls, _root, info, **data):
             info, data["product"], field="product", only_type=Product
         )
 
-        image_data = info.context.FILES.get(data["image"])
-        validate_image_file(image_data, "image")
+        image = data.get("image")
```

Comment:
IMO this should be placed in separate method, responsible only for checking/cleaning the input.